crates/viewer/re_renderer/shader/conversions/chroma_subsampling_converter.wgsl

```diff
@@ -0,0 +1,112 @@
+#import <../types.wgsl>
+#import <../screen_triangle_vertex.wgsl>
+
+struct UniformBuffer {
+    format: u32,
+    primaries: u32,
+    target_texture_size: vec2u,
+};
+
+@group(0) @binding(0)
+var<uniform> uniform_buffer: UniformBuffer;
+
+@group(0) @binding(1)
+var input_texture: texture_2d<u32>;
+
+
+const FORMAT_Y_UV = 0u;
+const FORMAT_YUYV16 = 1u;
+
+const PRIMARIES_BT601 = 0u;
+const PRIMARIES_BT709 = 1u;
+
+
+/// Returns sRGB from YUV color.
+///
+/// This conversion mirrors the function in `crates/store/re_types/src/datatypes/tensor_data_ext.rs`
+///
+/// Specifying the color standard should be exposed in the future [#3541](https://github.com/rerun-io/rerun/pull/3541)
+fn srgb_from_yuv(yuv: vec3f, primaries: u32) -> vec3f {
+    // rescale YUV values
+    let y = (yuv[0] - 16.0) / 219.0;
+    let u = (yuv[1] - 128.0) / 224.0;
+    let v = (yuv[2] - 128.0) / 224.0;
+
+    var rgb: vec3f;
+
+    switch (primaries) {
```

Comment:
wat, wgsl supports `switch`!? NICE! That could be useful in `crates/viewer/re_renderer/shader/rectangle_fs.wgsl` too 😄

Reply:
learned that today as well 😄. Followed a hunch there - probably I've seen it in the meanwhile somewhere.

I think it didn't back when we started. But Naga & Tint (Dawn's compiler) are happy with it so full steam ahead!